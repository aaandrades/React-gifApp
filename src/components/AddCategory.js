import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = (e) => {
    setinputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2 && inputValue !== "Enter your gif") {
      setCategories((values) => [inputValue, ...values]);
      setinputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        placeholder="Enter your gif"
        onChange={handleInputChange}
        className="input-container"
        spellCheck="false"
      />
      <button className="button-container" onClick={handleSubmit}>
        Search
      </button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
