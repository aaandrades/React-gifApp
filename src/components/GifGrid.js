import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data: images } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.prototype = { 
  category: PropTypes.string.isRequired,
}
