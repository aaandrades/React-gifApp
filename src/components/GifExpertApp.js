import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Death Note"]);

  // const handleAdd = () => {
  //   setCategories([...categories, "Death Note"]);
  // };

  return (
    <div>
      <h2> Gif Expert App </h2>
      <AddCategory setCategories={setCategories} />
      <p className="Paragraph">Search the gif that you want and press enter!</p>
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid 
              key={category} 
              category = {category} />
        ))}
      </ol>
    </div>
  );
};
