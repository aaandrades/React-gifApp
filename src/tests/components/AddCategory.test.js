import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Test for AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change the input text', () => {
        const input = wrapper.find('input');
        const value = 'Hello World in input';
        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe( value );
    });
    
    test('should submit the input but doesnt show', () => {
        wrapper.find('form').simulate('submit',{ preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and clean input text', () => {
        const value = 'Death note';
        wrapper.find('input').simulate('change', { target : {value}});
        wrapper.find('form').simulate('submit',{ preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes( 1 );
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        const newValue = wrapper.find('input').props().value;
        expect(newValue).toBe('');
    });

});
