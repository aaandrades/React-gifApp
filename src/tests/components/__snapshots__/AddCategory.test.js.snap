// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for AddCategory component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input-container"
    onChange={[Function]}
    placeholder="Enter your gif"
    spellCheck="false"
    type="text"
    value=""
  />
  <button
    className="button-container"
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
